tap_nice_incontact/streams.py

```diff
@@ -604,6 +612,188 @@ def get_records(self,
                         for rec in results.get(self.data_key))
 
 
+class DataExtractionStream(IncrementalStream):
+    """
+    Base class for streams pulling data via the Data Extraction API endpoints
+    Docs: https://help.nice-incontact.com/content/recording/dataextractionapi.htm
+    :param client: The API client used extract records from the external source
+    """
+    replication_method = 'INCREMENTAL'
+    entity_name = ''
+
+    # amount of time to delay before polling again
+    poll_delay = 5
+    # number of seconds before timing out
+    poll_timeout = 300
+    default_period = 'days'
```

Comment:
should this be configurable through the config object?

Reply:
Created a `poll_settings` dictionary in the config that you can set

Reply:
`default_period` is already supported as a config setting